aiida/tools/graph/graph_traversers.py

```diff
@@ -0,0 +1,287 @@
+    :param bool create_forward: will traverse CREATE links in the forward direction.
+    :param bool call_calc_forward: will traverse CALL_CALC links in the forward direction.
+    :param bool call_work_forward: will traverse CALL_WORK links in the forward direction.
+    """
+    traverse_links = parse_traversal_rules(GraphTraversalRules.DELETE, **kwargs)
+    function_output = traverse_graph(
+        starting_pks,
+        get_links=False,
+        links_forward=traverse_links['forward'],
+        links_backward=traverse_links['backward']
+    )
+    return function_output['nodes']
+
+
+def get_nodes_links_export(starting_pks, **kwargs):
+    """
+    This function will return the set of all nodes that can be connected
+    to a list of initial nodes through any sequence of specified authorized
+    links and directions for export. This will also return the links and
+    the traversal rules parsed.
+
+    :type starting_pks: list or tuple
+    :param starting_pks: Contains the (valid) pks of the starting nodes (also accepts sets)..
+    :param bool input_calc_forward: will traverse INPUT_CALC links in the forward direction.
+    :param bool create_backward: will traverse CREATE links in the backward direction.
+    :param bool return_backward: will traverse RETURN links in the backward direction.
+    :param bool input_work_forward: will traverse INPUT_WORK links in the forward direction.
+    :param bool call_calc_backward: will traverse CALL_CALC links in the backward direction.
+    :param bool call_work_backward: will traverse CALL_WORK links in the backward direction.
+    """
+    from aiida import orm
+
+    traverse_links = parse_traversal_rules(GraphTraversalRules.EXPORT, **kwargs)
+    traverse_output = traverse_graph(
+        starting_pks,
+        get_links=True,
+        links_forward=traverse_links['forward'],
+        links_backward=traverse_links['backward']
+    )
+
+    # The exporter wants a list of dictionaries for every link.
+    # I create a utility dictionary for mapping pk to uuid.
+    if traverse_output['nodes']:
+        qbuilder = orm.QueryBuilder().append(
+            orm.Node,
+            project=('id', 'uuid'),
+            filters={'id': {
+                'in': traverse_output['nodes']
+            }},
+        )
+        pk_2_uuid_dict = dict(qbuilder.all())
+    else:
+        pk_2_uuid_dict = {}
+
+    # The set of tuples now has to be transformed to a list of dicts
+    links_uuid_list = [{
+        'input': pk_2_uuid_dict[link.source_id],
+        'output': pk_2_uuid_dict[link.target_id],
+        'label': link.link_label,
+        'type': link.link_type
+    } for link in traverse_output['links']]
+
+    function_output = {
+        'nodes': traverse_output['nodes'],
+        'links': links_uuid_list,
+        'rules': traverse_links['rules_applied']
+    }
+
+    return function_output
+
+
+def parse_traversal_rules(ruleset=GraphTraversalRules.DEFAULT, **kwargs):
```

Comment:
Maybe `validate_traversal_rules` is better, as validate implies parsing but not vice versa